spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitAdmin.java

```diff
@@ -523,8 +526,22 @@ private void declareExchanges(final Channel channel, final Exchange... exchanges
 
 			if (!isDeclaringDefaultExchange(exchange)) {
 				try {
-					channel.exchangeDeclare(exchange.getName(), exchange.getType(), exchange.isDurable(),
-						exchange.isAutoDelete(), exchange.getArguments());
+					if (exchange.isDelayed()) {
+						Map<String, Object> arguments = exchange.getArguments();
+						if (arguments == null) {
+							arguments = new HashMap<String, Object>();
+						}
+						else {
+							arguments = new HashMap<String, Object>(arguments);
+						}
+						arguments.put("x-delayed-type", exchange.getType());
+						channel.exchangeDeclare(exchange.getName(), ExchangeTypes.DELAYED, exchange.isDurable(),
+								exchange.isAutoDelete(), arguments);
+					}
+					else {
+						channel.exchangeDeclare(exchange.getName(), exchange.getType(), exchange.isDurable(),
+							exchange.isAutoDelete(), exchange.getArguments());
```

Comment:
Perhaps I can convince you that, since this is the most likely branch of the if statement, we don't need to extract these arguments to variables in 99.999% of the time :smile: